UnityProject/Assets/Scripts/Systems/Explosions/NodeTypes/ExplosionNode.cs

```diff
@@ -105,9 +105,13 @@ public virtual float DoDamage(Matrix matrix, float DamageDealt, Vector3Int v3int
 			foreach (var integrity in matrix.Get<Integrity>(v3int, true))
 			{
 				//Throw items
-				if (integrity.GetComponent<ItemAttributesV2>() != null)
+				if (integrity.TryGetComponent<ItemAttributesV2>(out var traits))
 				{
-					integrity.GetComponent<UniversalObjectPhysics>().NewtonianPush(AngleAndIntensity.Rotate90(), 9,  1,3 ,  BodyPartType.Chest,integrity.gameObject, 15);
+					integrity.GetComponent<UniversalObjectPhysics>()?
+						.NewtonianPush(AngleAndIntensity.Rotate90(),
+							9,  1,3 ,
+							BodyPartType.Chest,integrity.gameObject, 15);
+					if (IgnoreAttributes != null && traits.GetTraits().Any(x => IgnoreAttributes.Contains(x))) continue;
```

Comment:
Might be worth converting this intoo a has any attribute function on ItemAttributesV2

Reply:
done